odl/diagnostics/operator.py

```diff
@@ -213,27 +214,40 @@ def adjoint(self):
         self._adjoint_definition()
         self._adjoint_of_adjoint()
 
-    def _derivative_convergence(self, step):
-        print('\nTesting derivative is linear approximation')
+    def _derivative_convergence(self):
+        name = 'Testing derivative is linear approximation'
 
-        with FailCounter("error = ||A(x+c*dx)-A(x)-c*A'(x)(dx)|| / "
-                         "|c| ||dx|| ||A||") as counter:
+        with FailCounter(name, "error = "
+                         "inf_c ||A(x+cp)-A(x)-A'(x)(cp)|| / c") as counter:
```

Comment:
This is definitely a better solution than the one before. Probably it's very hard to be more general than that.